devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+		
+		attribute "runningMode", "string"
+		attribute "setpointHold", "string"
+		attribute "prorgammingOperation", "string"
+		attribute "setpointHoldDuration", "number"
+		attribute "lockLevel", "number"
+		attribute "coolingDemand", "number"
+		attribute "heatingDemand", "number"
+		attribute "modelIdentifier", "string"
+		attribute "lastSeen","string"
+
+	}
+
+	preferences {
+		input ("lock_level", "enum", title: "Thermostat Keypad Lock Level", description: "Setting the Lock Level allows you to restric the on-device access.", options: ["Mode Only", "Setpoint","Full"], defaultValue: "Full")
+ 	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"thermostatFull", type:"thermostat", width:6, height:4) {
+			tileAttribute("device.temperature", key: "PRIMARY_CONTROL") {
+				attributeState("temp", label:'${currentValue}', unit:"dF", defaultState: true)
+			}
+			tileAttribute("device.temperature", key: "VALUE_CONTROL") {
+				attributeState("VALUE_UP", action: "tempUp")
+				attributeState("VALUE_DOWN", action: "tempDown")
+			}
+			tileAttribute("device.humidity", key: "SECONDARY_CONTROL") {
+				attributeState("humidity", label:'${currentValue}%', unit:"%", defaultState: true)
+			}
+			tileAttribute("device.thermostatOperatingState", key: "OPERATING_STATE") {
+				attributeState("idle", backgroundColor:"#00A0DC")
+				attributeState("heating", backgroundColor:"#e86d13")
+				attributeState("cooling", backgroundColor:"#00A0DC")
+			}
+			tileAttribute("device.thermostatMode", key: "THERMOSTAT_MODE") {
+				attributeState("off", label:'${name}')
+				attributeState("heat", label:'${name}')
+				attributeState("cool", label:'${name}')
+				attributeState("auto", label:'${name}')
+			}
+			tileAttribute("device.heatingSetpoint", key: "HEATING_SETPOINT") {
+				attributeState("heatingSetpoint", label:'${currentValue}', unit:"dF", defaultState: true)
+			}
+			tileAttribute("device.coolingSetpoint", key: "COOLING_SETPOINT") {
+				attributeState("coolingSetpoint", label:'${currentValue}', unit:"dF", defaultState: true)
+			}
+		}
+
+		valueTile("temperature", "device.temperature", width: 2, height: 2) {
+			state("temperature", label:'${currentValue}', unit:"dF",
+				backgroundColors:[
+							// Celsius
+							[value: 0, color: "#153591"],
+							[value: 7, color: "#1e9cbb"],
+							[value: 15, color: "#90d2a7"],
+							[value: 23, color: "#44b621"],
+							[value: 28, color: "#f1d801"],
+							[value: 35, color: "#d04e00"],
+							[value: 37, color: "#bc2323"],
+							// Fahrenheit
+							[value: 40, color: "#153591"],
+							[value: 44, color: "#1e9cbb"],
+							[value: 59, color: "#90d2a7"],
+							[value: 74, color: "#44b621"],
+							[value: 84, color: "#f1d801"],
+							[value: 95, color: "#d04e00"],
+							[value: 96, color: "#bc2323"]
+				]
+			)
+		}
+		standardTile("tempDown", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
```

Comment:
tempUp, tempDown are already covered by the multiAttributeTile, there shouldn't be a need to explicitly show these in separate tiles.

Reply:
On some mobile UI, target tap area for up/down buttons on multiAttributeTile is a 4mm circle vs 17mm square on a separate tile.  Even though, heat and cool setpoint tiles are present, it seems still useful to have a temp temp/down tile.

Reply:
Chose one, if separate tiles is preferred remove the up/down arrow from the multiAttributeTile. If separate tiles for heating and cooling setpoint is preferred this shouldn't be displayed at all.

Reply:
Actually, yes.  The separate heat and cool up/down can provide the alternate UI path.  Will remove the three tiles.

Reply:
extra up/down tiles removed.